### PR TITLE
perf(ci): speculative execution for pre-build-go-binaries

### PR DESCRIPTION
**Pull request by davdhacs**
### Problem

~42% of GHA runners are EPYC 7763 — older hardware with slower compute and significantly slower disk I/O. When a CI job lands on one, it takes 1.4-3.2x longer. Runner assignment is random with no way to retry to a specific type.

### Solution

Add `davdhacs/speedracer` — a composite action that runs multiple copies of a job in parallel. The first copy to land on a fast runner wins; losers cancel themselves within seconds.

**Usage** — add `speedracer: [a, b, ...]` as the last matrix dimension, call the action before checkout, :

\`\`\`yaml
strategy:
  matrix:
    arch: [amd64, arm64]
    speedracer: [a, b, c, d]
    exclude:
      - { speedracer: b, arch: arm64 }  # arm64 has consistent hardware
      - { speedracer: c, arch: arm64 }
      - { speedracer: d, arch: arm64 }
steps:
  - uses: davdhacs/speedracer@SHA # ratchet:davdhacs/speedracer@v0.1.0
  - uses: actions/checkout@...
  - ... actual work ...
\`\`\`

### Measured results

Across 4 verification runs on this PR:

| Job | Speedracer winner | If copy a hit 7763 (42%) | Saved |
|---|---|---|---|
| \`pre-build-go-binaries\` | 200-211s | ~380s | **169s (2.8 min)** |
| \`build-and-push-main\` | 142-195s | ~290s | **135s (2.2 min)** |
| \`build-and-push-scanner\` | 74-100s | ~270s | **187s (3.1 min)** |
| \`go\` unit tests | ~1366s | ~2100s | **730s (12.2 min)** |
| \`go-postgres\` | ~1372s | ~2060s | **688s (11.5 min)** |

When copy `a` naturally lands on a fast runner (58% of PRs), speedracer adds ~0s overhead — the winner runs at the same speed a single copy would have. The savings only apply to the 42% of PRs that would have hit a slow runner.

On the build critical path (\`pre-build -> build-and-push -> manifests\`), expected time drops from ~519s to ~408s average — **1.85 min faster on average**.

### When to apply speedracer

The value comes from disk I/O variance between runner CPU types. I/O-bound jobs (builds, image pushes, test suites with artifact uploads) benefit most. CPU-bound jobs (linting, style checks) see negligible benefit.

| Slow/fast ratio | Recommendation | Examples |
|---|---|---|
| **>1.5x** | Strong candidate | \`build-and-push-scanner\` (3.18x), \`pre-build-go-binaries\` (2.05x), \`build-and-push-main\` (1.87x) |
| **1.2-1.5x** | Apply if job is long | \`go\` tests (1.53x, 730s waste), \`go-postgres\` (1.50x, 688s waste) |
| **<1.2x** | Don't apply | \`check-generated-files\` (1.06x), \`style-check\` (1.02x) |

Copy count: each copy costs ~20s runner time (not blocking. It is worth spending more copies for longer jobs: A 1.5x ratio on a 23-minute job wastes 12 minutes if it runs on a slow runner — worth 4+ copies to avoid. A 3.2x ratio on a 1.5-minute job wastes 3 minutes — still worth 4+ copies at the cost of 3 extra ~20s runner slots.

## Testing and quality

- [x] CI results are inspected

### Automated testing

Comprehensive test suite in \`davdhacs/speedracer\` — 22 test scenarios + 11 API-verified downstream jobs + 4 TSV edge case tests + runner survey + arm64 verification. All 11 verify jobs pass.

### Observed behavior across 4 runs

All 4 runs across Build, Unit Tests, and Style workflows:
- Every speedracer job produced exactly one winner
- All losers show \`cancelled\` (not \`failure\`) — no fail-fast cascades
- Winners consistently run at fast-runner speed
- Cancelled copies exit in 15-27s
- Manifest push jobs all succeed
- arm64 copies unaffected (single copy, no speedracer overhead)

### CODE REVIEW

File: .github/workflows/style.yaml [hunk 1]
```diff
@@ -11,6 +11,7 @@ on:
     - opened
     - reopened
     - synchronize
+
```
Comment (davdhacs) [inline .github/workflows/style.yaml]:
```suggestion
```
